### PR TITLE
DOP-4622: Dark Mode - Button component

### PR DESCRIPTION
**Pull request by mmeigs**
### Stories/Links:

DOP-4622

### Current Behavior:

[Atlas homepage](https://www.mongodb.com/docs/atlas/)

### Staging Links:

[Atlas homepage](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/matt.meigs/DOP-4622-darkmode-button/index.html)
[Nested page where you can see the CollapsedBreadcrumb button with a small screen size](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/matt.meigs/DOP-4622-darkmode-button/index.html)
[Instruqt button](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/matt.meigs/DOP-4622-darkmode-button/atlas-search/tutorials/index.html)

### Notes:

- Mainly a version update fixed them all. I went through all instances, and none of our buttons have colors overridden.
- Deleted two unused components, too!

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: src/components/Sidenav/SidenavMobileMenuButton.js [hunk 1]
```diff
@@ -1,46 +0,0 @@
...
-  :focus {
-    color: ${palette.gray.dark3};
-  }
-
-  :before {
-    display: none;
-  }
-`;
-
-// TODO: Remove this component after consistent-nav is officially released
```
Comment (rayangler) [inline src/components/Sidenav/SidenavMobileMenuButton.js:25]: This is why it's good to file tickets instead of leaving TODOs 🙃 Thanks for cleaning up!

**Review by rayangler — APPROVED** [review]
LGTM!

File: src/components/Button.js [hunk 2]
```diff
@@ -37,7 +39,7 @@ const Button = ({
       className={cx(componentProps.as ? buttonStyling : '', 'button')}
       baseFontSize={baseFontSize}
       size={size}
-      darkMode={darkMode}
+      darkMode={darkModeProp ?? darkMode}
```
Comment (mayaraman19) [inline src/components/Button.js:42]: LGTM! just out of curiosity + to confirm, this prop does not override the provider, right?

**Review by mayaraman19 — APPROVED** [review]

File: src/components/Button.js (hunk 2 above)
Comment (mmeigs) [reply src/components/Button.js:42]: If it's true, yes!

Comment (mmeigs) [reply src/components/Button.js:42]: So, a button will never be light mode in a dark mode setting

**Pull request merged**